feat: [HL-51] Home New Recipe Horizontal View ✨

abdelhamid-f-nasser opened the pull request:
- Create Recipe model with properties, initializer, and formattedPreparationTime() method
- Implement RecipeTest XCTestCase subclass
- Add setUp(), tearDown() methods, and test cases for all properties and methods
- Include additional test cases for edge case scenarios
- Implement the cell for HomeNewRecipe

This implementation offers a structured representation of recipes and a comprehensive test suite to ensure correct behavior.

ticket: [HL-51] Home New Recipe View (Horizontal Collection View)

#### Proposed Changes

Implement the Home New Recipe view which shows the recipe item.

#### Screenshot

<img width="410" alt="Screenshot 2023-06-27 at 10 43 28 AM" src="https://github.com/motoon-eg/healthy/assets/38096011/a1a04717-ab7b-46f5-afc4-23905c332af8">

#### Checklist

- [x] I have tested these changes locally.
- [x] I have added appropriate documentation or updated existing documentation.
- [ ] I have added appropriate test coverage or updated existing test coverage.
- [ ] I have updated the changelog (if applicable).
- [x] I have followed the project's code style and formatting guidelines.
- [x] I have reviewed and adhered to the project's contributing guidelines.

<!--
#### Icons Catalog

- 🍕 Feature
- ✅ Test
-->


[HL-51]: https://motoon-eg.atlassian.net/browse/HL-51?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

abdelhamid-f-nasser committed:
feat: Add Recipe model and unit tests ✨

- Create Recipe model with properties, initializer, and formattedPreparationTime() method
- Implement RecipeTest XCTestCase subclass
- Add setUp(), tearDown() methods, and test cases for all properties and methods
- Include additional test cases for edge case scenarios

This implementation offers a structured representation of recipes and a comprehensive test suite to ensure correct behavior.

ticket: [HL-51] Home New Recipe View (Horizontal Collection View)

ahmdmhasn commented:
@abdelhamid-f-nasser Still in draft?

abdelhamid-f-nasser committed:
Merge branch 'develop' of https://github.com/motoon-eg/healthy into feat/HL-51-home-new-recipe-view

* 'develop' of https://github.com/motoon-eg/healthy:
  feat: [HL-65] Create search collection view cell 🍕🍕 (#74)
  feat: Method that create a title with a given view 🍕  (#65)
  test: [HL-44] Validator unit test (#64)
  chore: Add and integrate NewRelic 📈 (#69)
  feat: [HL-47] Header with current user icon.🍕🌟 (#56)

abdelhamid-f-nasser committed:
chore(Assets): Add New Recipe Item icons

Added new assets for the feature

abdelhamid-f-nasser committed:
feat(NewRecipe): Add new collection view cell

Add new recipe collection view cell

Note: Still need to implement rating bar, adjust card size

abdelhamid-f-nasser committed:
Merge branch 'develop' of https://github.com/motoon-eg/healthy into feat/HL-51-home-new-recipe-view

* 'develop' of https://github.com/motoon-eg/healthy:
  fix: [HL-30] Sign-In with Facebook ✨ (#78)

abdelhamid-f-nasser committed:
chore(Assets): Add Recipe Placeholder Image

abdelhamid-f-nasser committed:
refactor(NewRecipe): Adjust recipe cell design

- Add the recipe image on top of the card
- Adjust the card design and shadow

abdelhamid-f-nasser committed:
feat(UIView+Helpers): Add cornerRadius property

Add an @IBInspectable cornerRadius property to the
UIView+Helpers extension, allowing easy configuration of the corner
radius in Interface Builder or programmatically.

abdelhamid-f-nasser committed:
feat(NewRecipe): Add HomeNewRecipeCell

Add HomeNewRecipeCollectionViewCell and it's
associated assets.

ahmdmhasn commented on Healthy/Classes/Extensions/UIView+Helpers.swift:
```diff
@@ -30,4 +30,22 @@ extension UIView {
+                layer.cornerRadius = newValue
+                layer.masksToBounds = newValue > 0
+            }
+        }
```
```suggestion
    var cornerRadius: CGFloat {
        get {
            return layer.cornerRadius
        }
        set {
            layer.cornerRadius = newValue
            layer.masksToBounds = newValue > 0
        }
    }
```

ahmdmhasn commented on Healthy/Classes/Modules/Home/Views/HomeNewRecipesView/CollectionViewCell/HomeNewRecipeCollectionViewCell.swift:
```diff
@@ -0,0 +1,41 @@
+    @IBOutlet weak private(set) var userNameLabel: UILabel!
+    @IBOutlet weak private(set) var preparationTimeInMinutesLabel: UILabel!
+    @IBOutlet weak private(set) var preparationTimeHorizontalStackView: UIStackView!
+    @IBOutlet weak var contentBackgroundView: UIView!
```
```suggestion
    @IBOutlet private(set) weak var contentBackgroundView: UIView!
```

ahmdmhasn commented on Healthy/Classes/Modules/Home/Views/HomeNewRecipesView/CollectionViewCell/HomeNewRecipeCollectionViewCell.swift:
```diff
@@ -0,0 +1,41 @@
+    private func configureAppearance() {
+        recipeNameLabel.applyStyle(.textFieldTitleLabel)
+        userNameLabel.applyStyle(.signupSubtitle)
+//        preparationTimeInMinutesLabel.applyStyle(.signupSubtitle)
```
```suggestion
```

ahmdmhasn commented on Healthy/Classes/Modules/Home/Views/HomeNewRecipesView/CollectionViewCell/HomeNewRecipeCollectionViewCell.swift:
```diff
@@ -0,0 +1,41 @@
+        view.layer.masksToBounds = false
+    }
+
+}
```
```suggestion
    }
}
```

ahmdmhasn commented on Healthy/Classes/Modules/Home/Views/HomeNewRecipesView/CollectionViewCell/HomeNewRecipeCollectionViewCell.swift:
```diff
@@ -0,0 +1,41 @@
+import UIKit
+
+class HomeNewRecipeCollectionViewCell: UICollectionViewCell {
```
```swift
final class HomeNewRecipeCollectionViewCell: UICollectionViewCell {
    
    // MARK: - Outlets
    
    @IBOutlet weak private(set) var recipeNameLabel: UILabel!
    @IBOutlet weak private(set) var userImage: UIImageView!
    @IBOutlet weak private(set) var userNameLabel: UILabel!
    @IBOutlet weak private(set) var preparationTimeInMinutesLabel: UILabel!
    @IBOutlet weak private(set) var preparationTimeHorizontalStackView: UIStackView!
    @IBOutlet weak private(set) var contentBackgroundView: UIView!
    @IBOutlet weak private(set) var cardVerticalStackView: UIStackView!
    
    // MARK: - Lifecycle
    
    override func awakeFromNib() {
        super.awakeFromNib()
        // Initialization code
        configureAppearance()
    }
    
    override func layoutSubviews() {
        super.layoutSubviews()
        
        contentBackgroundView.applyDefaultCardShadow(cornerRadius: Constants.cardCornerRadius)
    }
    
    // MARK: - Private Methods
    
    private func configureAppearance() {
        recipeNameLabel.applyStyle(.textFieldTitleLabel)
        userNameLabel.applyStyle(.signupSubtitle)
        preparationTimeInMinutesLabel.applyStyle(.signupSubtitle)
    }
}

extension HomeNewRecipeCollectionViewCell {
    enum Constants {
        static let cardCornerRadius: CGFloat = 10.0
    }
}

extension UIView {
    func applyDefaultCardShadow(cornerRadius: CGFloat) {
        self.cornerRadius = cornerRadius
        backgroundColor = UIColor.white
        layer.shadowColor = UIColor.black.cgColor
        layer.shadowRadius = 4.0
        layer.shadowOpacity = 0.1
        layer.shadowOffset = .zero
        layer.masksToBounds = false
    }
}
```

ahmdmhasn commented on Healthy/Classes/Modules/Home/Views/HomeNewRecipesView/CollectionViewCell/HomeNewRecipeCollectionViewCell.xib:
```diff
@@ -0,0 +1,163 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
```xml
<?xml version="1.0" encoding="UTF-8"?>
<document type="com.apple.InterfaceBuilder3.CocoaTouch.XIB" version="3.0" toolsVersion="21507" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" useTraitCollections="YES" useSafeAreas="YES" colorMatched="YES">
    <device id="retina6_12" orientation="portrait" appearance="light"/>
    <dependencies>
        <deployment identifier="iOS"/>
        <plugIn identifier="com.apple.InterfaceBuilder.IBCocoaTouchPlugin" version="21505"/>
        <capability name="Named colors" minToolsVersion="9.0"/>
        <capability name="System colors in document resources" minToolsVersion="11.0"/>
        <capability name="documents saved in the Xcode 8 format" minToolsVersion="8.0"/>
    </dependencies>
    <objects>
        <placeholder placeholderIdentifier="IBFilesOwner" id="-1" userLabel="File's Owner"/>
        <placeholder placeholderIdentifier="IBFirstResponder" id="-2" customClass="UIResponder"/>
        <collectionViewCell opaque="NO" multipleTouchEnabled="YES" contentMode="center" id="gTV-IL-0wX" customClass="HomeNewRecipeCollectionViewCell" customModule="Healthy" customModuleProvider="target">
            <rect key="frame" x="0.0" y="0.0" width="350" height="243"/>
            <autoresizingMask key="autoresizingMask"/>
            <view key="contentView" opaque="NO" clipsSubviews="YES" multipleTouchEnabled="YES" contentMode="center">
                <rect key="frame" x="0.0" y="0.0" width="350" height="243"/>
                <autoresizingMask key="autoresizingMask" flexibleMaxX="YES" flexibleMaxY="YES"/>
                <subviews>
                    <view contentMode="scaleToFill" translatesAutoresizingMaskIntoConstraints="NO" id="pec-ZU-Obm" userLabel="Content View">
                        <rect key="frame" x="0.0" y="0.0" width="350" height="243"/>
                        <subviews>
                            <view contentMode="scaleToFill" translatesAutoresizingMaskIntoConstraints="NO" id="gb5-9c-VEW" userLabel="Card View">
                                <rect key="frame" x="0.0" y="40" width="350" height="203"/>
                                <subviews>
                                    <stackView contentMode="scaleToFill" axis="vertical" spacing="10" translatesAutoresizingMaskIntoConstraints="NO" id="5mh-gv-YjX" userLabel="Card Vertical Stack View">
                                        <rect key="frame" x="8" y="8" width="334" height="187"/>
                                        <subviews>
                                            <label opaque="NO" userInteractionEnabled="NO" contentMode="left" horizontalHuggingPriority="251" text="Steak With Tomoatoes" textAlignment="natural" lineBreakMode="tailTruncation" baselineAdjustment="alignBaselines" adjustsFontSizeToFit="NO" translatesAutoresizingMaskIntoConstraints="NO" id="lz8-Ww-lIv">
                                                <rect key="frame" x="0.0" y="0.0" width="334" height="121.66666666666667"/>
                                                <fontDescription key="fontDescription" type="system" pointSize="17"/>
                                                <nil key="textColor"/>
                                                <nil key="highlightedColor"/>
                                            </label>
                                            <label opaque="NO" userInteractionEnabled="NO" contentMode="left" horizontalHuggingPriority="251" verticalHuggingPriority="251" text="Label" textAlignment="natural" lineBreakMode="tailTruncation" baselineAdjustment="alignBaselines" adjustsFontSizeToFit="NO" translatesAutoresizingMaskIntoConstraints="NO" id="eSE-FA-QeZ" userLabel="Rating Bar">
                                                <rect key="frame" x="0.0" y="131.66666666666666" width="334" height="20.333333333333343"/>
                                                <fontDescription key="fontDescription" type="system" pointSize="17"/>
                                                <nil key="textColor"/>
                                                <nil key="highlightedColor"/>
                                            </label>
                                            <stackView opaque="NO" contentMode="scaleToFill" distribution="equalSpacing" translatesAutoresizingMaskIntoConstraints="NO" id="J51-Wy-SUt" userLabel="Bottom Horizontal Stack View">
                                                <rect key="frame" x="0.0" y="162" width="334" height="25"/>
                                                <subviews>
                                                    <stackView opaque="NO" contentMode="scaleToFill" spacing="10" translatesAutoresizingMaskIntoConstraints="NO" id="7Im-Mg-4Qz" userLabel="User Horizontal Stack View">
                                                        <rect key="frame" x="0.0" y="0.0" width="118.33333333333333" height="25"/>
                                                        <subviews>
                                                            <imageView clipsSubviews="YES" userInteractionEnabled="NO" contentMode="scaleToFill" horizontalHuggingPriority="251" verticalHuggingPriority="251" image="background-splash" translatesAutoresizingMaskIntoConstraints="NO" id="xGT-hy-wxA">
                                                                <rect key="frame" x="0.0" y="0.0" width="25" height="25"/>
                                                                <constraints>
                                                                    <constraint firstAttribute="width" constant="25" id="0sX-T6-Yew"/>
                                                                    <constraint firstAttribute="height" constant="25" id="edW-mU-AnV"/>
                                                                </constraints>
                                                                <userDefinedRuntimeAttributes>
                                                                    <userDefinedRuntimeAttribute type="number" keyPath="skeletonCornerRadius">
                                                                        <real key="value" value="0.0"/>
                                                                    </userDefinedRuntimeAttribute>
                                                                    <userDefinedRuntimeAttribute type="number" keyPath="cornerRadius">
                                                                        <real key="value" value="12.5"/>
                                                                    </userDefinedRuntimeAttribute>
                                                                </userDefinedRuntimeAttributes>
                                                            </imageView>
                                                            <label opaque="NO" userInteractionEnabled="NO" contentMode="left" horizontalHuggingPriority="251" verticalHuggingPriority="251" text="By James Miner" textAlignment="natural" lineBreakMode="tailTruncation" baselineAdjustment="alignBaselines" adjustsFontSizeToFit="NO" translatesAutoresizingMaskIntoConstraints="NO" id="RBy-SV-UY0">
                                                                <rect key="frame" x="34.999999999999993" y="0.0" width="83.333333333333314" height="25"/>
                                                                <fontDescription key="fontDescription" type="system" pointSize="11"/>
                                                                <color key="textColor" name="gray 3"/>
                                                                <nil key="highlightedColor"/>
                                                            </label>
                                                        </subviews>
                                                        <constraints>
                                                            <constraint firstAttribute="height" constant="25" id="tY7-0c-dCd"/>
                                                        </constraints>
                                                    </stackView>
                                                    <stackView opaque="NO" contentMode="scaleToFill" spacing="5" translatesAutoresizingMaskIntoConstraints="NO" id="EXg-Nh-LNb" userLabel="Preparation Time Horizontal Stack View">
                                                        <rect key="frame" x="270.33333333333331" y="0.0" width="63.666666666666686" height="25"/>
                                                        <subviews>
                                                            <imageView clipsSubviews="YES" userInteractionEnabled="NO" contentMode="scaleAspectFit" horizontalHuggingPriority="251" verticalHuggingPriority="251" image="icon-timer" translatesAutoresizingMaskIntoConstraints="NO" id="clN-Rs-zk1" userLabel="Preparation Time Icon">
                                                                <rect key="frame" x="0.0" y="0.0" width="17" height="25"/>
                                                            </imageView>
                                                            <label opaque="NO" userInteractionEnabled="NO" contentMode="left" horizontalHuggingPriority="251" verticalHuggingPriority="251" text="20 mins" textAlignment="natural" lineBreakMode="tailTruncation" baselineAdjustment="alignBaselines" adjustsFontSizeToFit="NO" translatesAutoresizingMaskIntoConstraints="NO" id="y0h-bh-RzJ" userLabel="Preparation Time In Minutes Label">
                                                                <rect key="frame" x="21.999999999999996" y="0.0" width="41.666666666666657" height="25"/>
                                                                <fontDescription key="fontDescription" type="system" pointSize="11"/>
                                                                <color key="textColor" name="gray 3"/>
                                                                <nil key="highlightedColor"/>
                                                            </label>
                                                        </subviews>
                                                        <constraints>
                                                            <constraint firstAttribute="height" constant="25" id="nzm-Ak-Mki"/>
                                                        </constraints>
                                                    </stackView>
                                                </subviews>
                                            </stackView>
                                        </subviews>
                                        <color key="backgroundColor" white="1" alpha="1" colorSpace="custom" customColorSpace="genericGamma22GrayColorSpace"/>
                                    </stackView>
                                </subviews>
                                <color key="backgroundColor" systemColor="systemBackgroundColor"/>
                                <constraints>
                                    <constraint firstItem="5mh-gv-YjX" firstAttribute="top" secondItem="gb5-9c-VEW" secondAttribute="top" constant="8" id="Pam-So-1SQ"/>
                                    <constraint firstAttribute="trailing" secondItem="5mh-gv-YjX" secondAttribute="trailing" constant="8" id="fu5-Fp-Iyb"/>
                                    <constraint firstAttribute="bottom" secondItem="5mh-gv-YjX" secondAttribute="bottom" constant="8" id="qJr-z8-RR9"/>
                                    <constraint firstItem="5mh-gv-YjX" firstAttribute="leading" secondItem="gb5-9c-VEW" secondAttribute="leading" constant="8" id="u2x-pC-Ihb"/>
                                </constraints>
                            </view>
                            <imageView clipsSubviews="YES" userInteractionEnabled="NO" contentMode="scaleAspectFit" horizontalHuggingPriority="251" verticalHuggingPriority="251" image="image-recipe-placeholder" translatesAutoresizingMaskIntoConstraints="NO" id="WpH-zC-qYH" userLabel="Recipe Image View">
                                <rect key="frame" x="270" y="0.0" width="80" height="80"/>
                                <constraints>
                                    <constraint firstAttribute="width" constant="80" id="feO-P3-uDg"/>
                                    <constraint firstAttribute="height" constant="80" id="lGz-d6-mh3"/>
                                </constraints>
                            </imageView>
                        </subviews>
                        <color key="backgroundColor" white="0.0" alpha="0.0" colorSpace="custom" customColorSpace="genericGamma22GrayColorSpace"/>
                        <constraints>
                            <constraint firstAttribute="trailing" secondItem="WpH-zC-qYH" secondAttribute="trailing" id="6Cd-Ri-Kc9"/>
                            <constraint firstItem="gb5-9c-VEW" firstAttribute="leading" secondItem="pec-ZU-Obm" secondAttribute="leading" id="a8D-P9-Uvx"/>
                            <constraint firstAttribute="trailing" secondItem="gb5-9c-VEW" secondAttribute="trailing" id="aNv-tT-2vr"/>
                            <constraint firstItem="gb5-9c-VEW" firstAttribute="top" secondItem="WpH-zC-qYH" secondAttribute="top" constant="40" id="hOq-uy-Lsi"/>
                            <constraint firstItem="WpH-zC-qYH" firstAttribute="top" secondItem="pec-ZU-Obm" secondAttribute="top" id="p1Y-23-XzT"/>
                            <constraint firstAttribute="bottom" secondItem="gb5-9c-VEW" secondAttribute="bottom" id="vTS-0z-6fV"/>
                        </constraints>
                    </view>
                </subviews>
            </view>
            <constraints>
                <constraint firstItem="pec-ZU-Obm" firstAttribute="top" secondItem="gTV-IL-0wX" secondAttribute="top" id="8gC-Ii-MqP"/>
                <constraint firstItem="pec-ZU-Obm" firstAttribute="leading" secondItem="gTV-IL-0wX" secondAttribute="leading" id="ItD-2h-vdn"/>
                <constraint firstAttribute="trailing" secondItem="pec-ZU-Obm" secondAttribute="trailing" id="svZ-MH-qeS"/>
                <constraint firstAttribute="bottom" secondItem="pec-ZU-Obm" secondAttribute="bottom" id="zcD-27-G5j"/>
            </constraints>
            <size key="customSize" width="1436" height="243"/>
            <connections>
                <outlet property="cardView" destination="gb5-9c-VEW" id="Ahj-YD-otc"/>
                <outlet property="contentBackgroundView" destination="pec-ZU-Obm" id="P5D-fZ-etl"/>
                <outlet property="preparationTimeHorizontalStackView" destination="EXg-Nh-LNb" id="iH2-ps-Q1p"/>
                <outlet property="preparationTimeInMinutesLabel" destination="y0h-bh-RzJ" id="w0e-Cn-n0j"/>
                <outlet property="recipeNameLabel" destination="lz8-Ww-lIv" id="7pM-oB-y6Y"/>
                <outlet property="userImage" destination="xGT-hy-wxA" id="uLm-yY-hrS"/>
                <outlet property="userNameLabel" destination="RBy-SV-UY0" id="p4m-by-kcQ"/>
            </connections>
            <point key="canvasLocation" x="957.25190839694653" y="-291.19718309859155"/>
        </collectionViewCell>
    </objects>
    <resources>
        <image name="background-splash" width="375" height="812"/>
        <image name="icon-timer" width="17" height="17"/>
        <image name="image-recipe-placeholder" width="120" height="112"/>
        <namedColor name="gray 3">
            <color red="0.66274509803921566" green="0.66274509803921566" blue="0.66274509803921566" alpha="1" colorSpace="custom" customColorSpace="sRGB"/>
        </namedColor>
        <systemColor name="systemBackgroundColor">
            <color white="1" alpha="1" colorSpace="custom" customColorSpace="genericGamma22GrayColorSpace"/>
        </systemColor>
    </resources>
</document>
```

ahmdmhasn commented on HealthyTests/Entities/RecipeTest.swift:
```diff
@@ -0,0 +1,132 @@
+        // Arrange
+        let actual = sut.title
+        let expectedTitle = "Steak with Tomatoes"
+        let testErrorMessage = "Recipe title is incorrect."
+        // Assert
```
```suggestion
        let testErrorMessage = "Recipe title is incorrect."
        
        // Assert
```

ahmdmhasn reviewed (approved):
🚀🚀🚀🚀🚀🚀

abdelhamid-f-nasser committed:
style(RecipeTest): Reformat file

Add separating lines between Arrange and Assert sections

abdelhamid-f-nasser committed:
refactor: make `cornerRadius` non-inspectable

Change `cornerRadius` to a non-IBInspectable property

abdelhamid-f-nasser committed:
refactor: Extract shadow to UIView extensions

- Move shadow method to UIView extensions
- Rename it to applyDefaultCardShadow
- Change it to accept cornerRadius
- Add documentation

abdelhamid-f-nasser committed:
refactor(NewRecipe): Modify Cell and bind data

- Modify Cell design in XIB to resolve crops and redundant constraints
- Configure the ViewModel binding data
- Refactor constants `cornerRadius` to constants enum.

ahmdmhasn commented on HealthyTests/Entities/RecipeTest.swift:
```diff
@@ -0,0 +1,142 @@
+
+        // Assert
+        XCTAssertEqual(actual, expectedFormattedPreparationTime, testErrorMessage)
+    }
+    func test_formattedPreparationTime_withMinutesOnly_ensureCorrectFormattedTime() {
```
```suggestion
    }

    func test_formattedPreparationTime_withMinutesOnly_ensureCorrectFormattedTime() {
```

ahmdmhasn commented on HealthyTests/Entities/RecipeTest.swift:
```diff
@@ -0,0 +1,142 @@
+
+        // Assert
+        XCTAssertEqual(actual, expectedFormattedPreparationTime, testErrorMessage)
+    }
+    func test_formattedPreparationTime_withValidPreparationTime_ensureCorrectFormattedTime() {
```
```suggestion
    }

    func test_formattedPreparationTime_withValidPreparationTime_ensureCorrectFormattedTime() {
```

ahmdmhasn commented on HealthyTests/Entities/RecipeTest.swift:
```diff
@@ -0,0 +1,142 @@
+        // Arrange
+        let actual = sut.userImageUrl
+        let userImageUrl = "https://example.com/user-avatar.jpg"
+        let testErrorMessage = "User image URL is incorrect."
+
+        // Assert
+        XCTAssertEqual(actual, userImageUrl, testErrorMessage)
+    }
+    func test_initializer_withValidUserName_ensureModelHasValidUserName() {
+        // Arrange
+        let actual = sut.userName
+        let userName = "John Doe"
+        let testErrorMessage = "User name is incorrect."
+
+        // Assert
+        XCTAssertEqual(actual, userName, testErrorMessage)
+    }
+    func test_initializer_withValidPreparationTime_ensureModelHasValidPreparationTime() {
+        // Arrange
+        let actual = sut.preparationTime
+        let expectedPreparationTime = 450
+        let testErrorMessage = "Recipe preparation time is incorrect."
+
+        // Assert
+        XCTAssertEqual(actual, expectedPreparationTime, testErrorMessage)
+    }
+    func test_initializer_withEmptyTitle_ensureModelHasEmptyTitle() {
+        // Arrange
+        let emptyTitleRecipe = Recipe(title: "",
+                                      recipeImageUrl: "https://example.com/test.jpg",
+                                      rating: 4,
+                                      userImageUrl: "https://example.com/user-avatar.jpg",
+                                      userName: "John Doe",
+                                      preparationTime: 300)
+        let actual = emptyTitleRecipe.title
+        let expectedTitle = ""
+        let testErrorMessage = "Recipe title with empty value is incorrect."
+
+        // Assert
+        XCTAssertEqual(actual, expectedTitle, testErrorMessage)
```
```suggestion
    // MARK: - Initializer Tests

    func test_initializer_withValidTitle_ensureModelHasValidTitle() {
        // Arrange
        let actual = sut.title
        let expectedTitle = "Steak with Tomatoes"
        let testErrorMessage = "Recipe title is incorrect."

        // Assert
        XCTAssertEqual(actual, expectedTitle, testErrorMessage)
    }

    func test_initializer_withValidImageUrl_ensureModelHasValidRecipeImageUrl() {
        // Arrange
        let actual = sut.recipeImageUrl
        let expectedImageUrl = "https://example.com/steak-with-tomatoes.jpg"
        let testErrorMessage = "Recipe image URL is incorrect."

        // Assert
        XCTAssertEqual(actual, expectedImageUrl, testErrorMessage)
    }

    func test_initializer_withValidRating_ensureModelHasValidRating() {
        // Arrange
        let actual = sut.rating
        let expectedRating = 4
        let testErrorMessage = "Recipe star rating is incorrect."

        // Assert
        XCTAssertEqual(actual, expectedRating, testErrorMessage)
    }

    func test_initializer_withValidUserImageUrl_ensureModelHasValidUserImageUrl() {
        // Arrange
        let actual = sut.userImageUrl
        let userImageUrl = "https://example.com/user-avatar.jpg"
        let testErrorMessage = "User image URL is incorrect."

        // Assert
        XCTAssertEqual(actual, userImageUrl, testErrorMessage)
    }

    func test_initializer_withValidUserName_ensureModelHasValidUserName() {
        // Arrange
        let actual = sut.userName
        let userName = "John Doe"
        let testErrorMessage = "User name is incorrect."

        // Assert
        XCTAssertEqual(actual, userName, testErrorMessage)
    }

    func test_initializer_withValidPreparationTime_ensureModelHasValidPreparationTime() {
        // Arrange
        let actual = sut.preparationTime
        let expectedPreparationTime = 450
        let testErrorMessage = "Recipe preparation time is incorrect."

        // Assert
        XCTAssertEqual(actual, expectedPreparationTime, testErrorMessage)
    }

    func test_initializer_withEmptyTitle_ensureModelHasEmptyTitle() {
        // Arrange
        let emptyTitleRecipe = Recipe(title: "",
                                      recipeImageUrl: "https://example.com/test.jpg",
                                      rating: 4,
                                      userImageUrl: "https://example.com/user-avatar.jpg",
                                      userName: "John Doe",
                                      preparationTime: 300)
        let actual = emptyTitleRecipe.title
        let expectedTitle = ""
        let testErrorMessage = "Recipe title with empty value is incorrect."

        // Assert
        XCTAssertEqual(actual, expectedTitle, testErrorMessage)
```

ahmdmhasn commented on Healthy/Classes/Modules/Home/Views/HomeNewRecipesView/CollectionViewCell/HomeNewRecipeCollectionViewCell.swift:
```diff
@@ -0,0 +1,89 @@
+        preparationTimeInMinutesLabel.text = with.preparationTimeInMinutes
+
+        // TODO: [HL-52] To be Implemented after implementing binding imageUrl To UIImageView
+//        recipeImage.image = with.recipeImageUrl
+//        userImage.image = with.userImageUrl
```
```suggestion
```

ahmdmhasn committed:
Update Healthy/Classes/Modules/Home/Views/HomeNewRecipesView/CollectionViewCell/HomeNewRecipeCollectionViewCell.swift

ahmdmhasn committed:
Update HealthyTests/Entities/RecipeTest.swift

ahmdmhasn committed:
Update HealthyTests/Entities/RecipeTest.swift

ahmdmhasn committed:
Update HealthyTests/Entities/RecipeTest.swift

abdelhamid-f-nasser committed:
Merge branch 'develop' of https://github.com/motoon-eg/healthy into feat/HL-51-home-new-recipe-view

* 'develop' of https://github.com/motoon-eg/healthy:
  Feat: [HL-70] Saved recipes table view cell 🍕. (#76)

# Conflicts:
#	Healthy/Resources/Generated/UIImage.Generated.swift

abdelhamid-f-nasser commented:
@ahmdmhasn 🐑 pleaseeeee

ahmdmhasn reviewed (approved):
🐑🐑🐑🐑❤️❤️❤️🎉

abdelhamid-f-nasser committed:
Merge branch 'develop' of https://github.com/motoon-eg/healthy into feat/HL-51-home-new-recipe-view

* 'develop' of https://github.com/motoon-eg/healthy:
  feat: [HL-33] Reusable loading view on button ♻️ (#75)
